pkgs/development/libraries/mesa/stubs.nix

```diff
@@ -1,50 +1,37 @@
 { stdenv
+, lib
 , libglvnd
-, mesa
 , OpenGL
 , testers
 }:
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "libGL";
-  inherit (if stdenv.hostPlatform.isDarwin then mesa else libglvnd) version;
+  version = if stdenv.hostPlatform.isDarwin then "4.1" else libglvnd.version;
   outputs = [ "out" "dev" ];
 
-  # On macOS, libglvnd is not supported, so we just use what mesa
-  # build. We need to also include OpenGL.framework, and some
-  # extra tricks to go along with. We add mesa’s libGLX to support
-  # the X extensions to OpenGL.
+  # On macOS, libglvnd is not supported, and mesa no longer builds (but it only
+  # provided a software renderer anyway). Provide the OpenGL framework as well
+  # as a pkg-config file for OpenGL.framework.
+  # GLX is not supported nor is OpenGL ES.
   buildCommand = if stdenv.hostPlatform.isDarwin then ''
-    mkdir -p $out/nix-support $dev
-    echo ${OpenGL} >> $out/nix-support/propagated-build-inputs
-    ln -s ${mesa.out}/lib $out/lib
+    mkdir -p $out/nix-support $dev/lib/pkgconfig $dev/nix-support
 
-    mkdir -p $dev/lib/pkgconfig $dev/nix-support
+    echo ${OpenGL} >> $out/nix-support/propagated-build-inputs
     echo "$out" > $dev/nix-support/propagated-build-inputs
-    ln -s ${mesa.dev}/include $dev/include
+
+    mkdir -p $out/lib
+    ln -s ${OpenGL}/Library/Frameworks/OpenGL.framework/OpenGL.tbd $out/lib/libGL.tbd
```

Comment:
From @reckenrode [on Matrix](https://matrix.to/#/!lheuhImcToQZYTQTuI:nixos.org/$XF4ope_PPZa-F1LWhhAkleN27fyXj6qAJSgfxvV9h8M?via=nixos.org&via=matrix.org&via=nixos.dev):

> Speaking of which, the Mesa libGL stuff should export the actual libGL.tbd not OpenGL.tbd. The latter also reeexports libGLU. I don’t know what kind of problems that could cause for apps thinking they can also link Mesa’s libGLU.
> …
> Correct. Under `OpenGL.framework/Versions/Current/Libraries` is `libGL.tbd` that exports only the OpenGL symbols.